knox/knox.sh

```diff
@@ -128,8 +159,31 @@ function initialize_crontab() {
 function install() {
   local master_name
   master_name=$(/usr/share/google/get_metadata_value attributes/dataproc-master)
-  if [[ "${master_name}" == $(hostname) ]]; then
+  if [[ "${DATAPROC_IMAGE_VERSION}" == "2.2" ]]; then
+      if [[ "${master_name}" == $(hostname -A | tr -d '[:space:]' | awk -F "." '{print $1}') ]]; then
+         [[ -z "${KNOX_GW_CONFIG_GCS}" ]] && err "Metadata knox-gw-config is not provided. knox-gw-config stores the configuration files for knox."
+          
+          retry_command "apt-get update"
+          install_dependencies
+          update
+          enable_demo_ldap_for_dev_testing
+          get_config_parameters
+          replace_the_master_key
+          generate_or_replace_certificate
+          initialize_crontab
+
+         [[ -f "/lib/systemd/system/knoxldapdemo.service" ]] && systemctl restart knoxldapdemo
+         systemctl restart knox
+      fi
+  elif [[ "${master_name}" == $(hostname) ]]; then
     [[ -z "${KNOX_GW_CONFIG_GCS}" ]] && err "Metadata knox-gw-config is not provided. knox-gw-config stores the configuration files for knox."
+      if [ "${DATAPROC_IMAGE_VERSION}" = "2.0" ]; then
+        echo "Dataproc version is 2.0"
+        SUB_MINOR_VERSION=$(get_dataproc_sub_minor_version)
+        if [ "${SUB_MINOR_VERSION}" -le 102 ]; then
+           apply_changes
```

Comment:
Update function name here, too.

Please include an issue link where the reason for this change is discussed, since it seems intended to patch an error.

Reply:
Done , issue link https://github.com/GoogleCloudDataproc/initialization-actions/issues/1157